falcon/util/uri.py

```diff
@@ -178,9 +178,8 @@ def decode(encoded_uri):
         tokens = decoded_uri.split('%')
         decoded_uri = tokens[0]
         for token in tokens[1:]:
-            char, byte = _HEX_TO_BYTE[token[:2]]
-            decoded_uri += char + token[2:]
-
+            char, byte = _HEX_TO_BYTE.get(token[:2], ('%', 0))
+            decoded_uri += char + (token[2:] if byte else token)
```

Comment:
Out of curiosity, I tested an alternative approach of using `try...except KeyError`, and found that it was slightly more efficient under Python 2.7. Would you mind making this change?